vertx-opentracing/src/main/java/io/vertx/tracing/opentracing/OpenTracingTracer.java

```diff
@@ -48,37 +46,35 @@ public OpenTracingTracer(boolean closeTracer, Tracer tracer) {
   }
 
   @Override
-  public <R> Scope receiveRequest(Context context, R request, String operation, Iterable<Map.Entry<String, String>> headers, TagExtractor<R> tagExtractor) {
+  public <R> Span receiveRequest(Context context, R request, String operation,
+    Iterable<Map.Entry<String, String>> headers, TagExtractor<R> tagExtractor) {
     SpanContext sc = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMap() {
       @Override
       public Iterator<Map.Entry<String, String>> iterator() {
         return headers.iterator();
       }
+
       @Override
       public void put(String key, String value) {
         throw new UnsupportedOperationException();
       }
     });
-    Tracer.SpanBuilder builder = tracer.buildSpan(operation);
-    if (sc != null) {
-      builder = builder.asChildOf(sc);
-    }
-    builder
+    Span span = tracer.buildSpan(operation)
       .ignoreActiveSpan()
+      .asChildOf(sc)
       .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
-      .withTag(Tags.COMPONENT.getKey(), "vertx");
-    Scope scope = builder.startActive(false);
-    reportTags(scope.span(), request, tagExtractor);
-    scope.close();
-    context.putLocal(ACTIVE_SCOPE, scope);
-    return scope;
+      .withTag(Tags.COMPONENT.getKey(), "vertx")
+      .start();
+    reportTags(span, request, tagExtractor);
+    context.putLocal(ACTIVE_SPAN, span);
```

Comment:
Use the `OpenTracingContext` class

Reply:
I didn't use the util class here, because `Vertx.currentContext()` uses the Context from the active `VertxThread`. In the case of this call, the Context that is being passed to the method is not attached to the current `VertxThread`. I ran into this issue when I was working on the ScopeManager implementation. It lead me to adding the `withContext` method, which was awkward at best. So for the OpenTracingTracer class, I used the context directly, and the Util class will be merely helper methods for "user" code.

Reply:
I see, you could have a helper method in the context utils class which accepts the context. I am fine with this if we expect folks will not need the same method in the utils.